### PR TITLE
Run tests on MacOS and Windows too

### PR DESCRIPTION
**Pull request by michael-k**

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -26,11 +35,15 @@ jobs:
...
+          python -m pip install --upgrade pip wheel setuptools tox tox-factor
+      - name: Set pyversion for tox
+        run: |
+          python -c "import sys; print('::set-output name=PYV::', ''.join([str(x) for x in sys.version_info[:2]]), sep='')"
+        id: pyversion
+      - name: Run tox targets for ${{ matrix.python-version }} on ${{ matrix.os }}
         run: |
-          ENV_PREFIX=$(tr -C -d "0-9" <<< "${{ matrix.python-version }}")
-          TOXENV=$(tox --listenvs | grep "^py$ENV_PREFIX" | tr '\n' ',') python -m tox
+          python -m tox -f py${{ steps.pyversion.outputs.PYV }}-${{ matrix.os }} -l
```
Comment (adamchainz) [inline .github/workflows/tests.yml:45]: I think we can remove the `-l` step - we implicitly see it based upon what actually runs?

File: tox.ini [hunk 2]
```diff
@@ -1,6 +1,7 @@
 [tox]
 envlist =
-    py{36,37,38,39}-twisted{187,latest}
+    py{36,37,38,39}-twisted{187,latest}-{ubuntu,macos}
+    py{36,37}-twistedlatest-windows
```
Comment (adamchainz) [inline tox.ini:4]: why the limit on twisted versions on windows? Also, might it be worth increasing the minimum twisted version first - 18.7 was released june 2018, and 20.3 is even more than 6 months old.

**Review by adamchainz — COMMENTED** [review]
Looking good

File: tox.ini (hunk 2 above)
Comment (michael-k) [reply tox.ini:4]: The tests fail if there's no wheel.  Which is the case for [18.7.0](https://pypi.org/project/Twisted/18.7.0/#files) and Python > 3.7.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (michael-k) [reply .github/workflows/tests.yml:45]: Sure; I've added that primarily for debugging; took me a while to figure out what works on all three OSes and what doesn't

I'm off for today; I'll remove it tomorrow :)

File: tox.ini (hunk 2 above)
Comment (adamchainz) [reply tox.ini:4]: Ah fair. Sounds like a reason to bump the minimum supported version 😁

Comment (michael-k) [reply tox.ini:4]: FYI: Twisted 19.7.0 is the first version that provides wheels for Python 3.6 and 3.7 on Windows: https://pypi.org/project/Twisted/19.7.0/#files

**Review by carltongibson — COMMENTED** [review]
Hi. So. What do we think about 🌳? 

I'm not sure of the value of running the whole matrix on all the platforms. 
Discuss. 😀

**Comment by michael-k** [comment]
Windows: Running the tests against Python 3.6 and 3.7 (once twisted has more wheels: 3.6 and 3.9) is probably good enough

MacOS: Not sure if anyone's using daphne on MacOS.  I've looked briefly over the issues but noticed nothing relevant

**Comment by carltongibson** [comment]
I run it to play, but there's nothing macOS specific about it. Same with Windows really. 

My inclination would be to a single run for each. 🧐

**Comment by carltongibson** [comment]
Hi @michael-k — I'm going to carry this on in #393. 

* I don't think there's anything macOS specific to test for. (So let's just add Windows for now, which **has** seen specific issues.
* I'll edge up the Twisted dependency to see what passes. I don't know that we need to support older Twisteds really... — but let's see. 
* I may cut it to just the one combination... (not sure yet) 

Thanks for this! 🏅

**Pull request closed without merging**